log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -165,6 +182,34 @@ private void processQueue()
 
             logMessageBatch(batch);
             batch.clear();
+
+            if (!retryBatch.isEmpty() && (retryThread == null || !retryThread.isAlive())) {
+                retryThread = new Thread(this::processRetryQueue, "log-retry");
+                retryThread.start();
+            }
+        }
+    }
+
+    private void processRetryQueue()
+    {
+        boolean isSuccess = false;
+        while (!isSuccess) {
+            isSuccess = logRetryMessageBatch();
+            if (isSuccess) {
+                retryBatch.trimToSize();
+                errorBackOff.success();
+                Thread.currentThread().interrupt();
```

Comment:
We are interrupting to finish on success? Why don't we just return from the task normally?